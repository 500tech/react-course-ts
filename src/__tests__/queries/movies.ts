import { print } from 'graphql/language/printer';
import {
  createMovie,
  fetchMovieDetails,
  fetchMovies,
  removeAllMovies,
  removeMovie,
  updateMovie
} from "../../queries/movies";

describe('Queries: Movies', () => {

  test('fetch movies query', () => {
    const result = print(fetchMovies);

    expect(result).toMatchSnapshot();
  });

  test('fetch movie details', () => {
    const result = print(fetchMovieDetails);

    expect(result).toMatchSnapshot();
  });

  test('create movie', () => {
    const result = print(createMovie);

    expect(result).toMatchSnapshot();
  });

  test('update movie', () => {
    const result = print(updateMovie);

    expect(result).toMatchSnapshot();
  });

  test('remove movie', () => {
    const result = print(removeMovie);

    expect(result).toMatchSnapshot();
  });

  test('remove all movies', () => {
    const result  = print(removeAllMovies(['0', '1', '2']));

    expect(result).toMatchSnapshot();
  })
});
