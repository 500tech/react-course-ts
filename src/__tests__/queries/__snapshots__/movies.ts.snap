// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries: Movies create movie 1`] = `
"fragment movie on Movie {
  id
  label
}

mutation createMovie($id: ID!, $label: String!, $rating: Int!, $description: String!) {
  createMovie(id: $id, label: $label, rating: $rating, description: $description) {
    ...movie
  }
}
"
`;

exports[`Queries: Movies fetch movie details 1`] = `
"query fetchMovieDetails($id: ID!) {
  Movie(id: $id) {
    description
    rating
  }
}
"
`;

exports[`Queries: Movies fetch movies query 1`] = `
"fragment movie on Movie {
  id
  label
}

query fetchMovies {
  allMovies {
    ...movie
  }
}
"
`;

exports[`Queries: Movies remove all movies 1`] = `
"mutation removeAllMovies {
  movie0: removeMovie(id: \\"0\\")
  movie1: removeMovie(id: \\"1\\")
  movie2: removeMovie(id: \\"2\\")
}
"
`;

exports[`Queries: Movies remove movie 1`] = `
"mutation removeMovie($id: ID!) {
  removeMovie(id: $id)
}
"
`;

exports[`Queries: Movies update movie 1`] = `
"fragment movie on Movie {
  id
  label
}

fragment fullMovie on Movie {
  ...movie
  rating
  description
}

mutation updateMovie($id: ID!, $label: String!, $rating: Int!, $description: String!) {
  updateMovie(id: $id, label: $label, rating: $rating, description: $description) {
    ...fullMovie
  }
}
"
`;
